code/modules/mob/living/silicon/ai/ai_say.dm

```diff
@@ -49,108 +49,103 @@
+	req.begin_async()
+	UNTIL(req.is_complete())
+	var/datum/http_response/res = req.into_response()
+	if(res.status_code == 200)
+		var/full_name_file = "data/vox_[vox_voice_number].wav"
+		shell("./data/ffmpeg.exe -nostats -loglevel 0 -i ./[full_name_file] -vn -y ./data/vox_[vox_voice_number].mp3")
+		if (!istype(SSassets.transport, /datum/asset_transport/webroot))
+			log_game("CDN not set up, VOX aborted.")
+			message_admins("CDN not set up, VOX aborted.")
+			fdel("data/vox_[vox_voice_number].wav")
+			fdel("data/vox_[vox_voice_number].mp3")
+			return
+		var/datum/asset_transport/webroot/WR = SSassets.transport
+		var/shit_fuck_ass = "data/vox_[vox_voice_number].mp3"
+		var/datum/asset_cache_item/ACI = new("[md5filepath(shit_fuck_ass)].mp3", file("data/vox_[vox_voice_number].mp3"))
+		ACI.namespace = "15aivox"
+		WR.save_asset_to_webroot(ACI)
+		var/url = WR.get_asset_url(null, ACI)
+
+		log_game("[key_name(speaker)] finished making a 15.AI announcement with the following message: [message]")
+		message_admins("[key_name(speaker)] finished making a 15.AI announcement with the following message: [message]")
+		speaker.say(";[message]")
+ 		// If there is no single listener, broadcast to everyone in the same z level
 		if(!only_listener)
 			// Play voice for all mobs in the z level
 			for(var/mob/M in GLOB.player_list)
-				if(M.can_hear() && (M.client.prefs.toggles & SOUND_ANNOUNCEMENTS))
-					var/turf/T = get_turf(M)
-					if(T.z == z_level)
-						SEND_SOUND(M, voice)
+				var/turf/T = get_turf(M)
+				if(T.z == z_level && M.can_hear())
+					to_chat(M, "<audio autoplay><source src=\"[url]\" type=\"audio/mpeg\"></audio>")
 		else
-			SEND_SOUND(only_listener, voice)
-		return TRUE
-	return FALSE
+			to_chat(only_listener, "<audio autoplay><source src=\"[url]\" type=\"audio/mpeg\"></audio>")
+		fdel("data/vox_[vox_voice_number].wav")
+		fdel("data/vox_[vox_voice_number].mp3")
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/delete_vox_statement, "[CONFIG_GET(string/asset_cdn_webroot)][WR.get_asset_suffex(ACI)]"), 30 SECONDS)
+		return 1
+	else
+		if(!res.status_code)
+			log_game("[key_name(speaker)] failed to produce a 15.AI announcement due to an error. Error: [req._raw_response]")
+			message_admins("[key_name(speaker)] failed to produce a 15.AI announcement due to an error. Error: [req._raw_response]")
+		else
+			log_game("[key_name(speaker)] failed to produce a 15.AI announcement due to an error. Error code: [res.status_code]")
+			message_admins("[key_name(speaker)] failed to produce a 15.AI announcement due to an error. Error code: [res.status_code]")
+		to_chat(speaker, "The speech synthesizer failed to return audio. Your speech cooldown has been reset. Please try again.")
+		fdel("data/vox_[vox_voice_number].wav")
+		fdel("data/vox_[vox_voice_number].mp3")
+		speaker.announcing_vox = world.time
+	return 0
+
+/proc/delete_vox_statement(string)
+	fdel(string)
```

Comment:
Make this a world proc so admins can't proc call it.

Reply:
done, it's /world/proc/delete_vox_statement now